src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/TimeSpanConverter.cs

```diff
@@ -0,0 +1,92 @@
+
+        public override TimeSpan Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(reader.TokenType);
+            }
+
+            bool isEscaped = reader._stringHasEscaping;
+            int maximumLength = isEscaped ? MaximumEscapedTimeSpanFormatLength : MaximumTimeSpanFormatLength;
+
+            ReadOnlySpan<byte> source = stackalloc byte[0];
+
+            if (reader.HasValueSequence)
+            {
+                ReadOnlySequence<byte> valueSequence = reader.ValueSequence;
+                long sequenceLength = valueSequence.Length;
+
+                if (!JsonHelpers.IsInRangeInclusive(sequenceLength, MinimumTimeSpanFormatLength, maximumLength))
+                {
+                    throw ThrowHelper.GetFormatException();
+                }
+
+                Span<byte> stackSpan = stackalloc byte[(int)sequenceLength];
+
+                valueSequence.CopyTo(stackSpan);
+                source = stackSpan;
+            }
+            else
+            {
+                source = reader.ValueSpan;
+
+                if (!JsonHelpers.IsInRangeInclusive(source.Length, MinimumTimeSpanFormatLength, maximumLength))
+                {
+                    throw ThrowHelper.GetFormatException();
+                }
+            }
+
+            if (isEscaped)
+            {
+                int backslash = source.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(backslash != -1);
+
+                Span<byte> sourceUnescaped = stackalloc byte[source.Length];
+
+                JsonReaderHelper.Unescape(source, sourceUnescaped, backslash, out int written);
+                Debug.Assert(written > 0);
+
+                source = sourceUnescaped.Slice(0, written);
+                Debug.Assert(!source.IsEmpty);
+            }
+
+            bool result = Utf8Parser.TryParse(source, out TimeSpan tmpValue, out int bytesConsumed, 'c');
+
+            // Note: Utf8Parser.TryParse will return true for invalid input so
+            // long as it starts with an integer. Example: "2021-06-18" or
+            // "1$$$$$$$$$$". We need to check bytesConsumed to know if the
+            // entire source was actually valid.
+
+            if (result && source.Length == bytesConsumed)
+            {
+                return tmpValue;
+            }
+
+            throw ThrowHelper.GetFormatException();
+        }
+
+        public override void Write(Utf8JsonWriter writer, TimeSpan value, JsonSerializerOptions options)
+        {
+            Span<byte> output = stackalloc byte[MaximumTimeSpanFormatLength];
+
+            bool result = Utf8Formatter.TryFormat(value, output, out int bytesWritten, 'c');
+            Debug.Assert(result);
+
+            writer.WriteStringValue(output.Slice(0, bytesWritten));
```

Comment:
[Non-blocking for this PR] When https://github.com/dotnet/runtime/pull/54254 goes in, we should consider updating this to use the new `Utf8JsonWriter.WriteRaw` API so we can avoid logic that checks whether the input payload needs to be escaped. We'd first need to quote the formatted value.